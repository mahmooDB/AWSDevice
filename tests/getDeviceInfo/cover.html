
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">device-db/tests/getDeviceInfo/main.go (80.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">// getDeviceInfo
package main

import (
        "encoding/json"
        "os"

        "data"

        "github.com/aws/aws-lambda-go/events"
        "github.com/aws/aws-lambda-go/lambda"
        "github.com/aws/aws-sdk-go/aws"
        "github.com/aws/aws-sdk-go/aws/session"
        "github.com/aws/aws-sdk-go/service/dynamodb"
        "github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
        //"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
)

// Handler: responsible for taking GET requests from user that provide a id by
// path parameter and produce appropriate response.
// Input: id (provided by GET/path)
// Output: json containing device information
func Handler(request events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) <span class="cov8" title="1">{

        // Try to get id from request
        id := request.PathParameters["id"]
        // If the input id is empty, return error 404
        if id == "" </span><span class="cov8" title="1">{
                return events.APIGatewayProxyResponse{
                        Body:       "No id provided!",
                        StatusCode: 404,
                }, nil
        }</span>

        // Add /devices/ path to the begining of input id
        <span class="cov8" title="1">id = "/devices/" + id

        // Initialize an aws session
        region := os.Getenv("AWS_REGION")
        sess, err := session.NewSession(&amp;aws.Config{
                Region: &amp;region},
        )
        // If somthing went wrong with session creation, return error 500
        if err != nil </span><span class="cov0" title="0">{
                return events.APIGatewayProxyResponse{
                        Body:       "Internal Server Error 1\nSession error: " + err.Error(),
                        StatusCode: 500,
                }, nil
        }</span>
        // Create DynamoDB client
        //db := dynamodb.New(sess)
        <span class="cov8" title="1">svc := dynamodb.New(sess)
        // Converting to mock DynamoDB client for test
        db := MockDynamoDB{svc}

        //Dyna := dynamodbiface.DynamoDBAPI(svc)

        // Get table name from OS
        tableName := aws.String(os.Getenv("DEVICES_TABLE_NAME"))
        // Try to get the requested item from DynamoDB table
        result, err := db.GetItem(&amp;dynamodb.GetItemInput{
                TableName: tableName,
                Key: map[string]*dynamodb.AttributeValue{
                        "id": {
                                S: aws.String(id),
                        },
                },
        })
        // If somthing went wrong with database, return error 500
        if err != nil </span><span class="cov0" title="0">{
                return events.APIGatewayProxyResponse{
                        Body:       "Internal Server Error 2\nDatabase error: " + err.Error(),
                        StatusCode: 500,
                }, nil
        }</span>
        // If the requested item was not found, return error 404
        <span class="cov8" title="1">if len(result.Item) == 0 </span><span class="cov8" title="1">{
                return events.APIGatewayProxyResponse{
                        Body:       "Not found!" + result.GoString(),
                        StatusCode: 404,
                }, nil
        }</span>

        // Create a device object from result
        <span class="cov8" title="1">device := data.Device{}
        err = dynamodbattribute.UnmarshalMap(result.Item, &amp;device)
        // If somthing went wrong with unmarshaling, return error 500
        if err != nil </span><span class="cov0" title="0">{
                return events.APIGatewayProxyResponse{
                        Body:       "Internal Server Error 3\nDatabase unmarshaling error: " + err.Error(),
                        StatusCode: 500,
                }, nil
        }</span>

        <span class="cov8" title="1">jsonResponse, err := json.Marshal(device)
        // If somthing went wrong with json creation, return error 500
        if err != nil </span><span class="cov0" title="0">{
                return events.APIGatewayProxyResponse{
                        Body:       "Internal Server Error 4\nJson marshaling error: " + err.Error(),
                        StatusCode: 500,
                }, nil
        }</span>

        // Finally, everything went smoothly! return requested device information.
        <span class="cov8" title="1">return events.APIGatewayProxyResponse{
                Body:       string(jsonResponse),
                StatusCode: 200,
        }, nil</span>

}

func main() <span class="cov0" title="0">{
        lambda.Start(Handler)
}</span>
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
