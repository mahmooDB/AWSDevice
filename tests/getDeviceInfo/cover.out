mode: set
device-db/getDeviceInfo/main.go:31.93,36.14 2 1
device-db/getDeviceInfo/main.go:44.2,53.16 4 1
device-db/getDeviceInfo/main.go:61.2,79.16 5 1
device-db/getDeviceInfo/main.go:86.2,86.27 1 1
device-db/getDeviceInfo/main.go:94.2,97.16 3 1
device-db/getDeviceInfo/main.go:104.2,106.16 2 1
device-db/getDeviceInfo/main.go:114.2,117.8 1 1
device-db/getDeviceInfo/main.go:36.14,41.3 1 1
device-db/getDeviceInfo/main.go:53.16,58.3 1 0
device-db/getDeviceInfo/main.go:79.16,84.3 1 0
device-db/getDeviceInfo/main.go:86.27,91.3 1 1
device-db/getDeviceInfo/main.go:97.16,102.3 1 0
device-db/getDeviceInfo/main.go:106.16,111.3 1 0
device-db/getDeviceInfo/main.go:121.13,123.2 1 0
